From: Mikulas Patocka <mpatocka@redhat.com>
Date: Thu, 12 Feb 2026 17:33:30 +0100
Subject: [PATCH] mm: allow __GFP_RETRY_MAYFAIL in vmalloc

The commit 07003531e03c8 ("mm/vmalloc: warn on invalid vmalloc gfp flags")
breaks the device mapper VDO target.  The VDO target calls vmalloc with
__GFP_RETRY_MAYFAIL and this flag is not in the mask of allowed flags.

There is no reason why vmalloc couldn't support __GFP_RETRY_MAYFAIL, so
let's add this flag to GFP_VMALLOC_SUPPORTED.

Link: https://lkml.kernel.org/r/ff48283b-be21-7f9a-d616-e303a4a1ebe6@redhat.com
Fixes: 07003531e03c ("mm/vmalloc: warn on invalid vmalloc gfp flags")
Signed-off-by: Mikulas Patocka <mpatocka@redhat.com>
Reported-by: Zdenek Kabelac <zkabelac@redhat.com>
Cc: Christoph Hellwig <hch@infradead.org>
Cc: Christoph Hellwig <hch@lst.de>
Cc: "Uladzislau Rezki (Sony)" <urezki@gmail.com>
Cc: SeongJae Park <sj@kernel.org>
Cc: <stable@vger.kernel.org>	[6.19]
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/vmalloc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index e286c2d2068cb..aea4b54e1e4bf 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -3928,6 +3928,7 @@ static void *__vmalloc_area_node(struct vm_struct *area, gfp_t gfp_mask,
  */
 #define GFP_VMALLOC_SUPPORTED (GFP_KERNEL | GFP_ATOMIC | GFP_NOWAIT |\
 				__GFP_NOFAIL |  __GFP_ZERO | __GFP_NORETRY |\
+				__GFP_RETRY_MAYFAIL |\
 				GFP_NOFS | GFP_NOIO | GFP_KERNEL_ACCOUNT |\
 				GFP_USER | __GFP_NOLOCKDEP)
 
-- 
2.47.3

